tests/v2/validation/snapshot/snapshot.go

```diff
@@ -115,49 +120,49 @@ func snapshotRestore(t *testing.T, client *rancher.Client, clusterName string, e
 	require.Equal(t, initialIngressName, ingressResp.ObjectMeta.Name)
 
 	if isRKE1 {
-		initialSnapshotValues := snapshotRKE1(t, client, podTemplate, deployment, clusterName, clusterID, localClusterID, etcdRestore, isRKE1)
-		restoreRKE1(t, client, initialSnapshotValues, etcdRestore, clusterName, clusterID)
+		restoreSnapshotResp, initialSnapshotValues := snapshotRKE1(t, client, podTemplate, deployment, clusterName, clusterID, localClusterID, etcdRestore, isRKE1)
+		restoreRKE1(t, client, initialSnapshotValues, etcdRestore, restoreSnapshotResp, clusterName, clusterID)
 	} else {
-		initialSnapshotValues := snapshotV2Prov(t, client, podTemplate, deployment, clusterName, clusterID, localClusterID, etcdRestore, isRKE1)
-		restoreV2Prov(t, client, initialSnapshotValues, etcdRestore, clusterName, clusterID)
+		restoreSnapshotConfig, initialSnapshotValues := snapshotV2Prov(t, client, podTemplate, deployment, clusterName, clusterID, localClusterID, etcdRestore, isRKE1)
+		restoreV2Prov(t, client, initialSnapshotValues, etcdRestore, restoreSnapshotConfig, clusterName, clusterID)
 	}
 
 	logrus.Infof("Deleting created workloads...")
 	err = steveclient.SteveType(DeploymentSteveType).Delete(deploymentResp)
 	require.NoError(t, err)
 
-	err = steveclient.SteveType(ServiceType).Delete(serviceResp)
+	err = steveclient.SteveType(serviceType).Delete(serviceResp)
 	require.NoError(t, err)
 
-	err = steveclient.SteveType(IngressSteveType).Delete(ingressResp)
+	err = steveclient.SteveType(ingressSteveType).Delete(ingressResp)
 	require.NoError(t, err)
 }
 
-func snapshotRKE1(t *testing.T, client *rancher.Client, podTemplate corev1.PodTemplateSpec, deployment *v1.Deployment, clusterName, clusterID, localClusterID string, etcdRestore *etcdsnapshot.Config, isRKE1 bool) initialSnapshotConfig {
+func snapshotRKE1(t *testing.T, client *rancher.Client, podTemplate corev1.PodTemplateSpec, deployment *v1.Deployment, clusterName, clusterID, localClusterID string, etcdRestore *etcdsnapshot.Config, isRKE1 bool) (*management.Cluster, initialSnapshotConfig) {
+	if etcdRestore.ReplaceNodes != (etcdsnapshot.ReplaceNodes{}) {
```

Comment:
not a fan of the way `(etcdsnapshot.ReplaceNodes{}` has to be called. Is there a less declarative way of doing this?

Reply:
Not that I'm aware of. Considering this is a struct, we can't just set it to some boolean or something like that. So this was the solution that consistently worked for me.

Reply:
Resolved offline